test/models/name_test.rb

```diff
@@ -2751,6 +2772,85 @@ def test_best_preferred_synonym
                  deprecated_name.best_preferred_synonym)
   end
 
+  # --------------------------------------
```

Comment:
Should this method and the following `test_dependency_...` methods be re-written as controller tests?
My thought was: 
- Test various permutations `dependency?` as model tests.
- Test the application of `dependency?` just once as a controller test.

Reply:
I totally agree with you -- testing Name#dependency? in model test.  We should probably be doing this sort of thing more often, but often it's just easier/lazier to take care of all the tests in one place.

The only thing I would avoid is testing of every level in a complex cascade of methods.  Like testing approved_synonym_of_proposed_name? and correctly_spelled_ancestor_of_proposed_name separately, then testing dependency?, too.  For the most part testing the public methods as block boxes makes sense, although there are exceptions -- internal parsers, for example, are good things to test directly even when private.